packages/cli/src/cmds/preview/task.ts

```diff
@@ -0,0 +1,87 @@
+import { ChildProcess, spawn, SpawnOptions } from "child_process";
+
+function delay(ms: number) {
+    // tslint:disable-next-line no-string-based-set-timeout
+    return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+interface TaskOptions {
+    cwd?: string;
+    env?: NodeJS.ProcessEnv,
+}
+
+export interface TaskResult {
+    success: boolean;
+    stdout: string;
+    stderr: string;
+    exitCode : number | null;
+}
+
+export class Task {
+    private task: ChildProcess;
+    private stdout: string = "";
+    private stderr: string = "";
+
+    constructor(command: string, options: TaskOptions) {
+        const spawnOptions: SpawnOptions = {
+            shell: true,
+            cwd: options.cwd,
+            env: options.env,
+        };
+        this.task = spawn(command, spawnOptions);
+        this.task.stdout?.on("data", (data) => {
+            // TODO: log
+            this.stdout += data;
+        });
+        this.task.stderr?.on("data", (data) => {
+            // TODO: log
+            this.stderr += data;
+        });
+    }
+
+    /**
+     * wait for the task to end
+     */
+    public async wait(): Promise<TaskResult> {
+        while (true) {
```

Comment:
Instead of `while(true)...delay`, can we leverage existing `exit` or `close` event? E.g.,

``` node
const task = spawn(...);
return new Promise<number>((resolve, reject) => {
    task.on('exit', code => resolve(code));
});
```

Reply:
Great idea! Moved `spawn` to `wait` and updated the implementation.